src/Admin/SiteHealth.php

```diff
@@ -20,18 +24,50 @@
  *
  * @since 1.5.0
  */
-final class SiteHealth {
+final class SiteHealth implements Service, Registerable, Delayed, Conditional {
+
+	/**
+	 * Service that monitors and controls the CSS transient caching.
+	 *
+	 * @var MonitorCssTransientCaching
+	 */
+	private $css_transient_caching;
+
+	/**
+	 * Check whether the conditional object is currently needed.
+	 *
+	 * @return bool Whether the conditional object is needed.
+	 */
+	public static function is_needed() {
+		return is_admin() && ! wp_doing_ajax();
+	}
+
+	/**
+	 * Get the action to use for registering the service.
+	 *
+	 * @return string Registration action to use.
+	 */
+	public static function get_registration_action() {
+		return 'wp_loaded';
+	}
+
+	/**
+	 * SiteHealth constructor.
+	 *
+	 * @param MonitorCssTransientCaching $css_transient_caching CSS transient caching monitoring service.
+	 */
+	public function __construct( MonitorCssTransientCaching $css_transient_caching ) {
+		$this->css_transient_caching = $css_transient_caching;
+	}
```

Comment:
Here's the auto-wiring dependency injector at play. Note that the typehint here in the construction is the only thing that ties these services together. There's no configuration needed in this basic case.

If we'd have an interface and one or more different implementations, we would use the interface here, and then define a `binding` in the `AmpWpPlugin` instead. Using interfaces in that way turns `AmpWpPlugin` into the central "dashboard" to define how the big building blocks are assembled.